### PR TITLE
CRM-19217 Ensure that is_active is set to 1 by default when creating …

### PR DESCRIPTION
**Pull request by seamuslee001**
…dashboards via api

@eileenmcnaughton following on your comment from the dashboard fix up PR of Coleman's

### CODE REVIEW

File: tests/phpunit/api/v3/DashboardTest.php [hunk 1]
```diff
@@ -59,6 +59,24 @@ public function testDashboardCreate() {
...
+    $this->assertEquals($dashboard['values'][$dashboard['id']]['is_active'], 1);
+  }
+
+  /**
+   * CRM-19217.
+   * Ensure that where is_active is specifically set to 0 is_active returns 0.
+   */
+  public function testDashboardCreateNotActive() {
+    $params = array(
+      'version' => 3,
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/DashboardTest.php]: callApiSuccess doesn't need you to set the version

Comment (seamuslee001) [inline tests/phpunit/api/v3/DashboardTest.php]: point will update :-) and tidy up other test as well

**Comment by eileenmcnaughton** [comment]
I'm giving this merge on pass - although it would be better to remove the version line I commented on this is definitely a good thing

File: api/v3/Dashboard.php [hunk 2]
```diff
@@ -61,6 +61,7 @@ function civicrm_api3_dashboard_create($params) {
  *   array of parameters determined by getfields.
  */
 function _civicrm_api3_dashboard_create_spec(&$params) {
+  $params['is_active']['api.default'] = 1;
   unset($params['version']);
```
Comment (seamuslee001) [inline api/v3/Dashboard.php]: @eileenmcnaughton WHY? (note i didn't add this in)

File: tests/phpunit/api/v3/DashboardTest.php (hunk 1 above)
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/DashboardTest.php]: If you are updating can you update the comment block - the full standard is there should be a line break after the first line

File: api/v3/Dashboard.php (hunk 2 above)
Comment (eileenmcnaughton) [inline api/v3/Dashboard.php]: no idea - seems like a mistake...

Comment (eileenmcnaughton) [inline api/v3/Dashboard.php]: maybe git blame will give a clue?

**Comment by colemanw** [comment]
@eileenmcnaughton I see you already merged this. My 2p would have been to just set this default in the schema. There's really no reason to have the sql column default to 0.

**Comment by eileenmcnaughton** [comment]
yes that would work too

**Pull request merged**